src/create_disk.sh

```diff
@@ -8,27 +8,65 @@
 # although see also https://github.com/coreos/coreos-assembler/pull/298
 # For people building "derived"/custom FCOS-like systems, feel free to file
 # an issue and we can discuss configuration needs.
-set -euo pipefail
 
-if [ "$#" -ne 8 ]; then
-	echo 'create_disk <device> <ostree-repo> <ostree-ref> <ostree-remote> <grub-script> <os-name> <save-var-subdirs> <space separated kargs>'
-	exit 1
-fi
+usage() {
+    cat <<EOC
+${0} create a supermin virtual machinge to create a
+Fedora CoreOS style disk image from an OS Tree.
+
+Options:
+    -R|--ostree-ref: the OSTRee reference to install
+    -d|--disk: disk device to use
+    -g|--grub-script: grub script to install
+    -h|--help: show this helper
+    -k|--kargs: kernel CLI args
+    -o|--osname: the OS name to use, e.g. fedora
+    -r|--ostree-remote: the ostree remote
+    -t|--ostree-repo: location of the ostree repo
+    -v|--save-var-subdirs: "yes" to workaround selabel issue for RHCOS
+
+You probably don't want to run this script by hand. This script is
+run as part of 'coreos-assembler build'.
+EOC
+}
+
+while [ $# -gt 0 ];
+do
+    flag="${1}"; shift;
+    arg="${1}"
```

Comment:
Would be cleaner to refer to the arg as `${1}` in the flag handling directly since e.g. we may add a flag that doesn't have an arg (technically, that's the case with `--save-var-subdirs` right now, we're just using `NONE` as a hack -- this PR will enable us to clean that up!).

Reply:
Cleaner, maybe. Readable, not so much, IMHO. FWIW, that's the reason the second `shift` happens after parsing for the use case you explained: having a boolean flag. 

It might be a pet-peeve of mine, but using `${1}` over the use of a more descriptive verb makes this a bit harder to grawk.

Reply:
Unconditionally doing `arg="${1}"` implies that every switch has an arg, which is true with today's args, but not true in the general case. As Andrew noticed, it also interferes with `set -u`. The other `getopt` loops in most of the codebase follow this pattern too (`shift`ing as part of the flag handling directly).

Reply:
Updated.